After 2025.9.0 warning: operation on 'raw_idx' may be undefined

### The problem

After updating with 2025.9.0 I'm seeing new warnings during the build.  This did not prevent the device from compiling.

```
src/esphome/components/sensirion_common/i2c_sensirion.cpp: In member function 'bool esphome::sensirion_common::SensirionI2CDevice::write_command_(uint16_t, esphome::sensirion_common::SensirionI2CDevice::CommandLen, const uint16_t*, uint8_t)':
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
   79 |     temp[raw_idx++] = crc8(&temp[raw_idx - 2], 2, 0xFF, CRC_POLYNOMIAL, true);
      |          ~~~~~~~^~
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
```

### Which version of ESPHome has the issue?

2025.9.0

### What type of installation are you using?

Docker

### What platform are you using?

ESP8266

### Component causing the issue

sensirion_common

### YAML Config

```yaml
# https://github.com/MallocArray/airgradient_esphome/blob/main/airgradient-basic.yaml
---
# AirGradient DIY Basic
# https://www.airgradient.com/open-airgradient/instructions/overview/

substitutions:
  name: "ag-basic"
  friendly_name: "AG Basic"
  name_add_mac_suffix: "false"  # Must have quotes around value

# Enable Home Assistant API
api:  # Add encryption key as desired

ota:  # Add password as desired
  platform: esphome
  # password:

wifi:

dashboard_import:
  package_import_url: github://MallocArray/airgradient_esphome/airgradient-basic.yaml
  import_full_config: false

packages:
  board: github://MallocArray/airgradient_esphome/packages/airgradient_d1_mini_board.yaml
  captive_portal: github://MallocArray/airgradient_esphome/packages/captive_portal.yaml
  pm_2.5: github://MallocArray/airgradient_esphome/packages/sensor_pms5003.yaml
  co2: github://MallocArray/airgradient_esphome/packages/sensor_s8.yaml
  # temp_humidity: github://MallocArray/airgradient_esphome/packages/sensor_sht40.yaml
  temp_humidity: github://MallocArray/airgradient_esphome/packages/sensor_sht30.yaml
  display: github://MallocArray/airgradient_esphome/packages/display_ssd1306.yaml
  airgradient_api: github://MallocArray/airgradient_esphome/packages/airgradient_api_d1_mini_no_sgp41.yaml
  wifi: github://MallocArray/airgradient_esphome/packages/sensor_wifi.yaml
  uptime: github://MallocArray/airgradient_esphome/packages/sensor_uptime.yaml
```

### Anything in the logs that might be useful for us?

```txt
INFO ESPHome 2025.9.0
INFO Reading configuration /config/release-ag-basic.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing ag-basic (board: d1_mini; framework: arduino; platform: platformio/espressif8266@4.2.1)
--------------------------------------------------------------------------------
Library Manager: Installing ESP32Async/ESPAsyncTCP @ 2.0.0
INFO Installing ESP32Async/ESPAsyncTCP @ 2.0.0
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ESPAsyncTCP@2.0.0 has been installed!
INFO ESPAsyncTCP@2.0.0 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
INFO Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.7.10 has been installed!
INFO ESPAsyncWebServer@3.7.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing bblanchon/ArduinoJson @ 7.4.2
INFO Installing bblanchon/ArduinoJson @ 7.4.2
Downloading  [####################################]  100%
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@7.4.2 has been installed!
INFO ArduinoJson@7.4.2 has been installed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
Dependency Graph
|-- ESPAsyncTCP @ 2.0.0
|-- ESP8266WiFi @ 1.0
|-- ESPAsyncWebServer @ 3.7.10
|-- ESP8266mDNS @ 1.2
|-- DNSServer @ 1.1.1
|-- Wire @ 1.0
|-- ArduinoJson @ 7.4.2
|-- ESP8266HTTPClient @ 1.2
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_frame_helper_plaintext.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_pb2.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_pb2_service.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/list_entities.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/proto.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/api/subscribe_state.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/button/button.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/captive_portal/captive_portal.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/copy/sensor/copy_sensor.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/dashboard_import/dashboard_import.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/display/display.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/display/display_buffer.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/display/rect.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/esp8266/core.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/esp8266/gpio.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/esp8266/helpers.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/esp8266/preferences.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/esphome/ota/ota_esphome.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/font/font.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/http_request/http_request.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/http_request/http_request_arduino.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/i2c/i2c.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/i2c/i2c_bus_arduino.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/json/json_util.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/logger/logger_esp8266.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/pmsx003/pmsx003.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/senseair/senseair.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/sensirion_common/i2c_sensirion.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/sensor/sensor.cpp.o
src/esphome/components/sensirion_common/i2c_sensirion.cpp: In member function 'bool esphome::sensirion_common::SensirionI2CDevice::write_command_(uint16_t, esphome::sensirion_common::SensirionI2CDevice::CommandLen, const uint16_t*, uint8_t)':
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
   79 |     temp[raw_idx++] = crc8(&temp[raw_idx - 2], 2, 0xFF, CRC_POLYNOMIAL, true);
      |          ~~~~~~~^~
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
Compiling .pioenvs/ag-basic/src/esphome/components/sht3xd/sht3xd.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/socket/lwip_raw_tcp_impl.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/socket/socket.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/ssd1306_base/ssd1306_base.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/ssd1306_i2c/ssd1306_i2c.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/switch/automation.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/switch/switch.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/template/sensor/template_sensor.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/template/switch/template_switch.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uart/uart.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uart/uart_component.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uart/uart_component_esp8266.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uart/uart_debugger.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uptime/sensor/uptime_seconds_sensor.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/uptime/sensor/uptime_timestamp_sensor.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/watchdog/watchdog.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/web_server/ota/ota_web_server.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/wifi/wifi_component_esp8266.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/components/wifi_signal/wifi_signal_sensor.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/application.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/color.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/component.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/controller.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/log.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/string_ref.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/time.cpp.o
Compiling .pioenvs/ag-basic/src/esphome/core/util.cpp.o
Compiling .pioenvs/ag-basic/src/main.cpp.o
Generating LD script .pioenvs/ag-basic/ld/local.eagle.app.v6.common.ld
Compiling .pioenvs/ag-basic/libaad/ESPAsyncTCP/AsyncPrinter.cpp.o
Compiling .pioenvs/ag-basic/libaad/ESPAsyncTCP/ESPAsyncTCP.cpp.o
Compiling .pioenvs/ag-basic/libaad/ESPAsyncTCP/ESPAsyncTCPbuffer.cpp.o
Compiling .pioenvs/ag-basic/libaad/ESPAsyncTCP/SyncClient.cpp.o
Compiling .pioenvs/ag-basic/libaad/ESPAsyncTCP/tcp_axtls.c.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/BearSSLHelpers.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/CertStoreBearSSL.cpp.o
Archiving .pioenvs/ag-basic/libaad/libESPAsyncTCP.a
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFi.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiAP.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiGeneric.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiGratuitous.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiMulti.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiSTA-WPS.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiSTA.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/ESP8266WiFiScan.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/WiFiClient.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/WiFiClientSecureBearSSL.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/WiFiServer.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/WiFiServerSecureBearSSL.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/WiFiUdp.cpp.o
Compiling .pioenvs/ag-basic/libfb1/ESP8266WiFi/enable_wifi_at_boot_time.cpp.o
Compiling .pioenvs/ag-basic/lib181/Hash/Hash.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncEventSource.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncJson.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncMessagePack.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncWebHeader.cpp.o
Archiving .pioenvs/ag-basic/lib181/libHash.a
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncWebServerRequest.cpp.o
Archiving .pioenvs/ag-basic/libfb1/libESP8266WiFi.a
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/AsyncWebSocket.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/BackPort_SHA1Builder.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/ChunkPrint.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/Middleware.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/WebAuthentication.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/WebHandlers.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/WebRequest.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/WebResponses.cpp.o
Compiling .pioenvs/ag-basic/lib2af/ESPAsyncWebServer/WebServer.cpp.o
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/ESP8266mDNS.cpp.o
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/LEAmDNS.cpp.o
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/LEAmDNS_Control.cpp.o
Archiving .pioenvs/ag-basic/lib2af/libESPAsyncWebServer.a
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/LEAmDNS_Helpers.cpp.o
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/LEAmDNS_Structs.cpp.o
Compiling .pioenvs/ag-basic/libfb2/ESP8266mDNS/LEAmDNS_Transfer.cpp.o
Compiling .pioenvs/ag-basic/lib070/DNSServer/DNSServer.cpp.o
Compiling .pioenvs/ag-basic/lib953/Wire/Wire.cpp.o
Compiling .pioenvs/ag-basic/lib29b/ESP8266HTTPClient/ESP8266HTTPClient.cpp.o
Archiving .pioenvs/ag-basic/lib953/libWire.a
Archiving .pioenvs/ag-basic/libFrameworkArduinoVariant.a
Compiling .pioenvs/ag-basic/FrameworkArduino/Crypto.cpp.o
Archiving .pioenvs/ag-basic/lib070/libDNSServer.a
Compiling .pioenvs/ag-basic/FrameworkArduino/Esp-frag.cpp.o
Archiving .pioenvs/ag-basic/libfb2/libESP8266mDNS.a
Compiling .pioenvs/ag-basic/FrameworkArduino/Esp-version.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Esp.cpp.o
Archiving .pioenvs/ag-basic/lib29b/libESP8266HTTPClient.a
Compiling .pioenvs/ag-basic/FrameworkArduino/FS.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/FSnoop.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/HardwareSerial.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/IPAddress.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/LwipDhcpServer-NonOS.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/LwipDhcpServer.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/LwipIntf.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/LwipIntfCB.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/MD5Builder.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Print.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Schedule.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/StackThunk.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Stream.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/StreamSend.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Tone.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/TypeConversion.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/Updater.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/WMath.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/WString.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/abi.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/aes_unwrap.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/base64.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/cbuf.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/cont.S.o
Compiling .pioenvs/ag-basic/FrameworkArduino/cont_util.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_app_entry_noextra4k.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_eboot_command.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_features.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_flash_quirks.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_flash_utils.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_i2s.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_main.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_non32xfer.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_noniso.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_phy.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_postmortem.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_si2c.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_sigma_delta.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_spi_utils.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_timer.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_vm.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_waveform_phase.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_waveform_pwm.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring_analog.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring_digital.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring_pulse.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring_pwm.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/core_esp8266_wiring_shift.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/crc32.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/debug.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/exc-c-wrapper-handler.S.o
Compiling .pioenvs/ag-basic/FrameworkArduino/exc-sethandler.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/flash_hal.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/gdb_hooks.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/heap.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/hwdt_app_entry.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/libb64/cdecode.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/libb64/cencode.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/libc_replacements.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/mmu_iram.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/reboot_uart_dwnld.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs/spiffs_cache.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs/spiffs_check.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs/spiffs_gc.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs/spiffs_hydrogen.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs/spiffs_nucleus.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/spiffs_api.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/sqrt32.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/stdlib_noniso.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/time.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/uart.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/umm_malloc/umm_info.c.o
Compiling .pioenvs/ag-basic/FrameworkArduino/umm_malloc/umm_integrity.c.o
Compiling .pioenvs/ag-basic/FrameworkArduino/umm_malloc/umm_local.c.o
Compiling .pioenvs/ag-basic/FrameworkArduino/umm_malloc/umm_malloc.cpp.o
Compiling .pioenvs/ag-basic/FrameworkArduino/umm_malloc/umm_poison.c.o
Compiling .pioenvs/ag-basic/FrameworkArduino/wpa2_eap_patch.cpp.o
Archiving .pioenvs/ag-basic/libFrameworkArduino.a
Linking .pioenvs/ag-basic/firmware.elf
RAM:   [=====     ]  45.7% (used 37460 bytes from 81920 bytes)
Flash: [=====     ]  54.2% (used 565761 bytes from 1044464 bytes)
Building .pioenvs/ag-basic/firmware.bin
esp8266_copy_factory_bin([".pioenvs/ag-basic/firmware.bin"], [".pioenvs/ag-basic/firmware.elf"])
esp8266_copy_ota_bin([".pioenvs/ag-basic/firmware.bin"], [".pioenvs/ag-basic/firmware.elf"])
======================== [SUCCESS] Took 349.87 seconds ========================
INFO Successfully compiled program.
```

### Additional information

_No response_

Warning present in ESPHome 2025.9.0 and 2025.9.1.
Likely introduced/caused by https://github.com/esphome/esphome/pull/10604

```
INFO ESPHome 2025.9.1
INFO Reading configuration /config/esphome/dom-balkon.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing esp-balkon (board: esp01_1m; framework: arduino; platform: platformio/espressif8266@4.2.1)
--------------------------------------------------------------------------------
Library Manager: Installing ESP32Async/ESPAsyncTCP @ 2.0.0
INFO Installing ESP32Async/ESPAsyncTCP @ 2.0.0
Unpacking  [####################################]  100%
Library Manager: ESPAsyncTCP@2.0.0 has been installed!
INFO ESPAsyncTCP@2.0.0 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
INFO Installing ESP32Async/ESPAsyncWebServer @ 3.7.10
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.7.10 has been installed!
INFO ESPAsyncWebServer@3.7.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/noise-c @ 0.1.10
INFO Installing esphome/noise-c @ 0.1.10
Unpacking  [####################################]  100%
Library Manager: noise-c@0.1.10 has been installed!
INFO noise-c@0.1.10 has been installed!
Library Manager: Resolving dependencies...
INFO Resolving dependencies...
Library Manager: Installing esphome/libsodium @ 1.10020.7
INFO Installing esphome/libsodium @ 1.10020.7
Unpacking  [####################################]  100%
Library Manager: libsodium@1.10020.7 has been installed!
INFO libsodium@1.10020.7 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 7.4.2
INFO Installing bblanchon/ArduinoJson @ 7.4.2
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@7.4.2 has been installed!
INFO ArduinoJson@7.4.2 has been installed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 1MB Flash
Dependency Graph
|-- ESPAsyncTCP @ 2.0.0
|-- ESP8266WiFi @ 1.0
|-- ESPAsyncWebServer @ 3.7.10
|-- ESP8266mDNS @ 1.2
|-- noise-c @ 0.1.10
|-- Wire @ 1.0
|-- ArduinoJson @ 7.4.2
Compiling .pioenvs/esp-balkon/src/esphome/components/api/api_connection.cpp.o
Compiling .pioenvs/esp-balkon/src/esphome/components/api/api_frame_helper.cpp.o
Compiling .pioenvs/esp-balkon/src/esphome/components/api/api_frame_helper_noise.cpp.o
....... <cut> .......
Compiling .pioenvs/esp-balkon/src/esphome/components/sensor/sensor.cpp.o
src/esphome/components/sensirion_common/i2c_sensirion.cpp: In member function 'bool esphome::sensirion_common::SensirionI2CDevice::write_command_(uint16_t, esphome::sensirion_common::SensirionI2CDevice::CommandLen, const uint16_t*, uint8_t)':
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
   79 |     temp[raw_idx++] = crc8(&temp[raw_idx - 2], 2, 0xFF, CRC_POLYNOMIAL, true);
      |          ~~~~~~~^~
src/esphome/components/sensirion_common/i2c_sensirion.cpp:79:17: warning: operation on 'raw_idx' may be undefined [-Wsequence-point]
Compiling .pioenvs/esp-balkon/src/esphome/components/sht3xd/sht3xd.cpp.o
....... <cut> .......
Compiling .pioenvs/esp-balkon/FrameworkArduino/umm_malloc/umm_poison.c.o
Compiling .pioenvs/esp-balkon/FrameworkArduino/wpa2_eap_patch.cpp.o
Archiving .pioenvs/esp-balkon/libFrameworkArduino.a
Linking .pioenvs/esp-balkon/firmware.elf
RAM:   [====      ]  44.0% (used 36072 bytes from 81920 bytes)
Flash: [=====     ]  52.3% (used 535885 bytes from 1023984 bytes)
Building .pioenvs/esp-balkon/firmware.bin
esp8266_copy_factory_bin([".pioenvs/esp-balkon/firmware.bin"], [".pioenvs/esp-balkon/firmware.elf"])
esp8266_copy_ota_bin([".pioenvs/esp-balkon/firmware.bin"], [".pioenvs/esp-balkon/firmware.elf"])
======================== [SUCCESS] Took 459.39 seconds ========================
INFO Successfully compiled program.
................
```

--- a/esphome/components/sensirion_common/i2c_sensirion.cpp
+++ b/esphome/components/sensirion_common/i2c_sensirion.cpp
@@ -76,7 +76,8 @@ bool SensirionI2CDevice::write_command_(uint16_t command, CommandLen command_len
     temp[raw_idx++] = data[i] >> 8;
 #endif
     // Use MSB first since Sensirion devices use CRC-8 with MSB first
-    temp[raw_idx++] = crc8(&temp[raw_idx - 2], 2, 0xFF, CRC_POLYNOMIAL, true);
+    uint8_t crc = crc8(&temp[raw_idx - 2], 2, 0xFF, CRC_POLYNOMIAL, true);
+    temp[raw_idx++] = crc;
   }
   this->last_error_ = this->write(temp, raw_idx);
   return this->last_error_ == i2c::ERROR_OK;

